File: lib/domain/pest/entities/crop_pest_link_entity.rb
# frozen_string_literal: true

module Domain
  module Pest
    module Entities
      class CropPestLinkEntity
        attr_reader :id, :crop_id, :pest_id

        def initialize(id:, crop_id:, pest_id:)
          @id = id
          @crop_id = crop_id
          @pest_id = pest_id
        end
      end
    end
  end
end

File: crates/agrr-domain/src/pest/entities/crop_pest_link_entity.rs
/// Ruby: `Domain::Pest::Entities::CropPestLinkEntity`
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub struct CropPestLinkEntity {
    pub id: i64,
    pub crop_id: i64,
    pub pest_id: i64,
}

impl CropPestLinkEntity {
    pub fn new(id: i64, crop_id: i64, pest_id: i64) -> Self {
        Self {
            id,
            crop_id,
            pest_id,
        }
    }
}
